src/roles/actors.rs

```diff
@@ -0,0 +1,356 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use rstd::prelude::*;
+use parity_codec::Codec;
+use parity_codec_derive::{Encode, Decode};
+use srml_support::{StorageMap, StorageValue, dispatch, decl_module, decl_storage, decl_event, ensure, Parameter};
+use srml_support::traits::{Currency, EnsureAccountLiquid};
+use runtime_primitives::traits::{Zero, Bounded, SimpleArithmetic, As, Member, MaybeSerializeDebug, MaybeDebug};
+use system::{self, ensure_signed};
+use crate::governance::{GovernanceCurrency, BalanceOf };
+
+use crate::traits::{Members, Roles};
+
+#[derive(Encode, Decode, Copy, Clone, Eq, PartialEq, Debug)]
+pub enum Role {
+    Storage,
+}
+
+
+#[cfg_attr(feature = "std", derive(Debug))]
+#[derive(Encode, Decode, Copy, Clone, Eq, PartialEq)]
+pub struct RoleParameters<T: Trait> {
+    // minium balance required to stake to enter a role
+    pub min_stake: BalanceOf<T>,
+
+    // the maximum number of spots available to fill for a role
+    pub max_actors: u32,
+
+    // minimum actors to maintain - if role is unstaking
+    // and remaining actors would be less that this value - prevent or punish for unstaking
+    pub min_actors: u32,
+
+    // fixed amount of tokens paid to actors' primary account
+    pub reward: BalanceOf<T>,
+
+    // payouts are made at this block interval
+    pub reward_period: T::BlockNumber,
+
+    // how long tokens remain locked for after unstaking
+    pub unbonding_period: T::BlockNumber,
+
+    // minimum amount of time before being able to unstake
+    pub bonding_time: T::BlockNumber,
```

Comment:
The name looks inconsistent with `reward_period ` and `unbonding_period `
```suggestion
    pub bonding_period: T::BlockNumber,
```

Reply:
yep caught that last night and fixed.